src/NHibernate/Loader/DefaultEntityAliases.cs

```diff
@@ -66,85 +66,40 @@ protected virtual string[] GetIdentifierAliases(ILoadable persister, string suff
 
 		protected virtual string[] GetPropertyAliases(ILoadable persister, int j)
 		{
-			return persister.GetPropertyAliases(suffix, j);
+			return persister.GetPropertyAliases(_suffix, j);
 		}
 
-		private string[] GetUserProvidedAliases(string propertyPath, string[] defaultAliases)
+		private string[] DetermineKeyAliases(ILoadable persister)
 		{
-			string[] result = propertyPath == null ? null : GetUserProvidedAlias(propertyPath);
-			if (result == null)
-			{
-				return defaultAliases;
-			}
-			else
-			{
-				return result;
-			}
+			return
+				GetUserProvidedAlias(persister.IdentifierPropertyName) ??
+				GetUserProvidedAlias(EntityPersister.EntityID) ??
+				GetIdentifierAliases(persister, _suffix);
 		}
 
-		private string[] GetUserProvidedAlias(string propertyPath)
+		private string[][] DeterminePropertyAliases(ILoadable persister)
 		{
-			string[] result;
-			userProvidedAliases.TryGetValue(propertyPath, out result);
-			return result;
+			return GetSuffixedPropertyAliases(persister);
 		}
 
-		private string GetUserProvidedAlias(string propertyPath, string defaultAlias)
+		private string DetermineDiscriminatorAlias(ILoadable persister)
 		{
-			string[] columns = propertyPath == null ? null : GetUserProvidedAlias(propertyPath);
-			if (columns == null)
-			{
-				return defaultAlias;
-			}
-			else
+			string[] columns = GetUserProvidedAlias(AbstractEntityPersister.EntityClass);
+			if (columns != null)
```

Comment:
The last refactoring removed by the way a discrepancy in previous refactoring causing a `null` value as first `columns` value to be replaced. So the last refactoring fix an undue change. (That said, this likely concerns an error case.)

Reply:
I don't get it.

Reply:
The previous state of the PR was somewhat equivalent to
```c#
return columns?[0] ?? GetDiscriminatorAlias(persister, _suffix);
```

While the master code was more like current state.

So on master, if `columns[0]` contains `null`, `null` is yielded as an alias (which is likely an error case). WIth your last commit it will also be the case. But without your last commit, it would have been replaced by `GetDiscriminatorAlias(persister, _suffix)`.

Reply:
I believe that this is a correct behaviour: if we got null in `columns[0]` we should not swallow this case, but fail somewhere.

Reply:
Yes, agreed.